examples/default/flake.nix

```diff
@@ -1,36 +1,59 @@
 {
-  description = "Fleek Configuration";
-
   inputs = {
-    # Nixpkgs
+    # Principle inputs (updated by `nix run .#update`)
     nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
-
-    # Home manager
     home-manager.url = "github:nix-community/home-manager";
     home-manager.inputs.nixpkgs.follows = "nixpkgs";
+
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    nixos-flake.url = "github:bketelsen/nixos-flake";
+    fleek.url = "github:ublue-os/fleek";
   };
 
-  outputs = { nixpkgs, home-manager, ... }@inputs: {
+  outputs = inputs@{ self, ... }:
+    let
+      inherit (self) outputs;
+    in
+    inputs.flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = [ "x86_64-linux" "aarch64-linux" "aarch64-darwin" "x86_64-darwin" ];
 
-    # Available through 'home-manager --flake .#your-username@your-hostname'
-    homeConfigurations = {
-    
-      "bjk@beast" = home-manager.lib.homeManagerConfiguration {
-        pkgs = nixpkgs.legacyPackages.x86_64-linux; # Home-manager requires 'pkgs' instance
-        extraSpecialArgs = { inherit inputs; }; # Pass flake inputs to our config
-        modules = [ 
-          ./home.nix 
-          ./path.nix
-          ./shell.nix
-          ./user.nix
-          ./aliases.nix
-          ./programs.nix
-          # Host Specific configs
-          ./beast/beast.nix
-          ./beast/user.nix
+      imports = [
+        inputs.nixos-flake.flakeModule
+        ./users
+      ];
+      perSystem = { self', pkgs, lib, config, inputs', ... }:
+        {
+          legacyPackages.homeConfigurations."${self.people.users.bjk}@beast" =
```

Comment:
[Interesting](https://github.com/bketelsen/nixos-flake/commit/35f0e1c2d1c4ead8fc346078ce9c86392708e8c4)

But what if the user wants to share the same configuration across multiple hosts? I suppose they can explicitly assign it as such:

```nix
{
  legacyPackages.homeConfigurations = rec {
    default = ...;
    "srid@macbook1" = default;
    "srid@linux-desktop1" = default;
  };
}
```

Hmm, perhaps `nixos-flake` should be changed to support this convention.

Reply:
technically users do share the same configuration across all their hosts, but fleek gives them an escape hatch for user-specific and host-specific configs.